overlord/devicestate/handlers_serial.go

```diff
@@ -388,6 +390,50 @@ func prepareSerialRequest(t *state.Task, regCtx registrationContext, privKey ass
 		return "", retryErr(t, nTentatives, "cannot read response with request-id for making a request for a serial: %v", err)
 	}
 
+	st.Lock()
+	var hasPrepareSerialRequestHook bool
+	err = st.Get("has-prepare-serial-request-hook", &hasPrepareSerialRequestHook)
+	st.Unlock()
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		return "", err
+	}
```

Comment:
We can just get the gadget info and check for the hook here instead of writing flags into the global state